WordPress/Classes/Services/ReaderTopicService+Subscriptions.swift

```diff
@@ -107,18 +81,37 @@ extension ReaderTopicService {
 
 
 extension ReaderTopicService {
-    @nonobjc public func subscribeSiteNotifications(with siteId: NSNumber, _ success: @escaping () -> Void, _ failure: @escaping (NSError?) -> Void) {
-        toggleSiteNotifications(with: siteId, subscribe: true, success, failure)
-    }
+    /// Toggle site notifications subscription for new post
+    ///
+    /// - Parameters:
+    ///   - siteId: Site id to be used
+    ///   - subscribe: Flag to define is subscribe or unsubscribe
+    ///   - success: Success block
+    ///   - failure: Failure block
+    func toggleSubscribingNotifications(for siteId: Int?, subscribe: Bool, _ success: (() -> Void)? = nil, _ failure: ((NSError?) -> Void)? = nil) {
+        guard let siteId = siteId else {
+            let error = NSError(domain: "ReaderTopicService", code: 0, userInfo: [NSLocalizedDescriptionKey: "The site id is nil"])
```

Comment:
Where we're creating Errors to pass to failure blocks, it might be nice if we could abstract the error creation so we could both encapsulate it, and to make it easier to add new errors in the future should the need arise.  You might take a peek at `SnapshotError` in `WordPress/WordPressScreenshotGeneration/SnapshotHelper.swift` for an example of one approach. 

Additionally, if we're using `NSLocalizedDescriptionKey` then the message should be localized as well.

Reply:
Perfect! I'll modify also The remote service using the same Error rather than NSError.